coders/webp.c

```diff
@@ -862,53 +921,107 @@ static MagickBooleanType WriteAnimatedWEBPImage(const ImageInfo *image_info,
   WebPAnimEncoderOptions
     enc_options;
 
-  WebPData
-    webp_data;
-
   WebPPicture
     picture;
 
   WebPAnimEncoderOptionsInit(&enc_options);
   if (image_info->verbose)
     enc_options.verbose = 1;
-
-  image=CoalesceImages(image, exception);
-  first_image=image;
   enc=WebPAnimEncoderNew((int) image->page.width,(int) image->page.height,
     &enc_options);
 
-  head=(PictureMemory *) calloc(sizeof(*head),1);
-  current=head;
+  first_image=CoalesceImages(image,exception);
```

Comment:
The reason that image was assigned is because `ThrowWriterException` will call `DestroyImage`. It would probably be a good idea to have a single variable for the `while` loop. Or to use `ThrowBinaryException` and `break` from the loop. This way we we also call `WebPAnimEncoderDelete` that was already a bug in the code before you made your patches.

Reply:
Thinks for review, i change `ThrowWriterException ` to `ThrowMagickException ` and fix some uncatch error from libwebp.